macro to read TOF DCS entries

chiarazampolli opened the pull request.

chiarazampolli committed:
macro to read TOF DCS entries

clang

clang again

adding includes only when CLING and ROOT_CLING are not defined

Removing ifdef for one of the macros

removing unnecessary includes

chiarazampolli commented:
This should be a trivial change, just the addition of a macro. @shahor02 , I think @noferini  would not complain if you approve it, since it changes nothing in the TOF code.

chiarazampolli commented:
Hello @TimoWilken , @ktf , 

I am confused... I ran clang-format and nothing was meant to be changed:

```
[O2/latest] ~/SOFT/alibuild/ali-o2/O2 $> git clang-format
no modified files to format
[O2/latest] ~/SOFT/alibuild/ali-o2/O2 $>
```

I thought that the CI and what one gets from alibuild was the same... 
Chiara

shahor02 reviewed (dismissed):
@chiarazampolli do you have locally the same clang-format 11 as running in CI? This should be the reason of the discrepancy.

chiarazampolli commented:
It was just a trivial mistake, all is fine. Locally I could reproduce it. I noticed after I added the comment (now deleted). 
Sorry for the noise.
And thanks @shahor02 for approving!

chiarazampolli commented:
Error in dataflow seems unrelated.

shahor02 commented on Detectors/TOF/calibration/macros/readTOFDCSentries.C:
```diff
@@ -0,0 +1,62 @@
+// macro to read the TOF DCS information from CCDB
+// default ts is very big: Saturday, November 20, 2286 5:46:39 PM
+
+#include "CCDB/CcdbApi.h"
+#include "DetectorsDCS/AliasExpander.h"
+#include "DetectorsDCS/DeliveryType.h"
+#include "DetectorsDCS/DataPointIdentifier.h"
+#include "TOFCalibration/TOFDCSProcessor.h"
+
+#include <string>
+#include <unordered_map>
+#include <chrono>
+#include <bitset>
```
Could you wrap these includes in ``#if !defined(__CLING__) || defined(__ROOTCLING__)`` / ``#endif``

chiarazampolli replied:
Done

chiarazampolli replied:
Hello @shahor02 , 

Was it not all includes that were needed in the ifdef?

shahor02 replied:
No, the ``DetectorsDCS/AliasExpander.h`` is not a class known to ROOTCLING via dictionary but just a declaration of simple functions, should be visible also to ROOTCLING.
Actually, my comment related to other macro which failed in the prev. CI test, I just put it in a wrong place...

chiarazampolli replied:
I don't understand why the "read" macro failed, if the "make" macro did not when I committed it the first time. I thought they should be the same in this respect, and the "make" macro was not failing before, why should the "read" do? So I treated them in the same way. Unless there are some specific "includes" in the "read" one.

Anyway, now the "make" does not contain the "ifdef" anymore, only the "read".

chiarazampolli replied:
I also cleaned up the includes now.
